otherlibs/systhreads/st_stubs.c

```diff
@@ -52,22 +52,60 @@
 #include "caml/sys.h"
 #include "caml/memprof.h"
 
-#include "../../runtime/sync_posix.h"
-
 /* "caml/threads.h" is *not* included since it contains the _external_
    declarations for the caml_c_thread_register and caml_c_thread_unregister
    functions. */
 
 /* Max computation time before rescheduling, in milliseconds */
 #define Thread_timeout 50
 
+typedef int st_retcode;
+
+/* OS-specific initialization */
+static int st_initialize(void)
+{
+  return 0;
+}
+
 /* OS-specific code */
 #ifdef _WIN32
 #include "st_win32.h"
 #else
 #include "st_posix.h"
 #endif
 
+
+static uintnat st_masterlock_waiters(st_masterlock * m)
+{
+  return atomic_load_acquire(&m->waiters);
+}
+
+static void st_bt_lock_acquire(st_masterlock *m) {
+
+  /* We do not want to signal the backup thread if it is not "working"
+     as it may very well not be, because we could have just resumed
+     execution from another thread right away. */
+  if (caml_bt_is_in_blocking_section()) {
+    caml_bt_enter_ocaml();
+  }
+
+  caml_acquire_domain_lock();
+}
```

Comment:
Pedantically, if these were supposed to be moves:
```suggestion

  return;
}
```